packages/@lwc/template-compiler/src/__tests__/fixtures/svg/scoped-id-dynamic/expected.js

```diff
@@ -2,74 +2,66 @@ import _implicitStylesheets from "./scoped-id-dynamic.css";
 import _implicitScopedStylesheets from "./scoped-id-dynamic.scoped.css?scoped=true";
 import {
   freezeTemplate,
-  parseSVGFragment,
+  parseFragment,
   registerTemplate,
   sanitizeAttribute,
 } from "lwc";
-const $fragment1 = parseSVGFragment`<defs${3}><circle${"a1:id"} r="10" cx="10" cy="10" fill="black"${3}/><circle${"a2:id"} r="10" cx="14" cy="14" fill="red"${3}/></defs>`;
-const stc0 = {
-  attrs: {
-    width: "100px",
-    height: "100px",
-    viewport: "0 0 100 100",
-  },
-  key: 0,
-  svg: true,
-};
+const $fragment1 = parseFragment`<svg width="100px" height="100px" viewport="0 0 100 100"${3}><defs${3}><circle${"a2:id"} r="10" cx="10" cy="10" fill="black"${3}/><circle${"a3:id"} r="10" cx="14" cy="14" fill="red"${3}/></defs><use${"a4:href"}${3}/><use${"a5:xlink:href"}${3}/></svg>`;
```

Comment:
Will using `parseSVGFragment` vs `parseFragment` cause any observable changes?

I'm not sure if I'm interpreting this correctly but it looks like with [`parseSVGFragment`](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/engine-core/src/framework/template.ts#L358) the content that's returned is just the `<defs>` whereas with `parseFragment` the content will be returned with a wrapping `<svg>` tag.

If it does, will it break any existing tests?

Reply:
This is expected. For the `<svg>` element itself, we always do `parseFragment`, not `parseSVGFragment`, due to this logic:

https://github.com/salesforce/lwc/blob/17c990d4f6b3986bad6c5870851fb62d48aa4f6d/packages/%40lwc/template-compiler/src/codegen/codegen.ts#L625-L628

The reason this changed in the above fixture is because we are static-optimizing the whole `<svg>` rather than the `<defs>` now.

I think the reason we did it this way is because `<svg>` automatically forces the HTML parser to use the SVG namespace, but for the stuff inside, we have to explicitly explain that it's in the SVG namespace.

Reply:
[Here's a CodePen to demonstrate](https://codepen.io/nolanlawson-the-selector/pen/QWRerKd?editors=1010). Note that the `<circle>` only works when parsed within the `<svg>`, not by itself.